### PR TITLE
Kinematic grab

### PR DESCRIPTION
**Pull request by sethalves**
- add a new action type (kinematic-hold) which sets position rather than velocity
- update handControllerGrab.js to use kinematic-hold

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6a770a870b98be3d9120d50fcfcd3ad0fd7fc026 available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26310d8ce421369193be808b451cf09d7418c787 available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8c25f2e16c9ccea98ce03a1520160f11e0837e33 available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8f924f0be45be2386c3f4fc9a92459dd4985587e available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7f8411d947a714e11d7a3aceb04f8cb2b58d5fc8 available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 99be44f92d77dfc18c4d1a7678bc2157c39534df available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 91f5a6fec6b79a43e2a1013a04b0e07dc9531057 available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

File: interface/src/avatar/AvatarActionKinematicHold.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    if (!_mine) {
+        // if a local script isn't updating this, then we are just getting spring-action data over the wire.
+        // let the super-class handle it.
+        ObjectActionSpring::updateActionWorker(deltaTimeStep);
+        return;
+    }
+
+    if (deltaTimeStep <= 0.0f) {
+        return;
+    }
```
Comment (AndrewMeadows) [inline interface/src/avatar/AvatarActionKinematicHold.cpp]: **deltaTimeStep** will never be <= 0.0

You can assert that is true here.

**Comment by hifi-gustavo** [comment]
Build console output for commit: febdb52b267d82bfb48dbb173c8aa789c49e968f available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 18b5b994b4e41e8f8cfa1b43173320000560facb available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5abaf38efd6acb314dd29b78a953ab26cfaf7be7 available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8caa7abca67d7a15856ecffec1cdbd52cd37ee21 available at: https://gist.github.com/753aec122673b6c50fff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5986.dmg

**Pull request merged**